src/Operator.ts

```diff
@@ -111,6 +111,7 @@ export default class Operator {
   /**
    * Create a copy of the vector.
    * @todo Make it more lightweight than using lodash.
+   * FIXME: Is this operation even necessary? Operator should never be mutated anyway.
    */
   copy(): Operator {
     return _.cloneDeep(this)
```

Comment:
Good question.
I think it may be needed in some simulations, where we create a new, separate state of our quantum world.

Reply:
If operator was to be made readonly (and i don't really see why that would be a bad idea), then copy is not necessary in any case. If you want to create a slightly different version of existing one, just go through the constructor as we do in all of the methods. If you agree, I would like to remove that.

Reply:
Within the current code, yes - I think that you are right and `readonly` is the best solution.